### ISSUE TITLE
Can't install lupa with luajit x86 & python x64

### ISSUE THREAD

**Issue by AutumnSun1996** [scoder/lupa#200]
I need a lupa build for 64bit machine which can execute 32bit luajit bytecode.

I'm able to build a working luajit, but can't install lupa with it.

The error message says 
```
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_err.o)' is incompatible with i386:x86-64 output
```

<summary>
This is full messages.
<details>

```
 > [stage-0 9/9] RUN cd lupa-master     && python setup.py bdist_wheel --lua-includes ../LuaJIT-2.1.0-beta3/src --lua-lib ../LuaJIT-2.1.0-beta3/src/libluajit.a:
#14 3.444 Using Lua library: ../LuaJIT-2.1.0-beta3/src/libluajit.a
#14 3.444 Using Lua include directory: ../LuaJIT-2.1.0-beta3/src
#14 3.444 generated sources not available, need Cython to build
#14 3.444 building with Cython 0.29.26
#14 3.444 Compiling /lupa-master/lupa/_lupa.pyx because it changed.
#14 3.444 [1/1] Cythonizing /lupa-master/lupa/_lupa.pyx
#14 3.888 running bdist_wheel
#14 3.892 running build
#14 3.892 running build_py
#14 4.033 creating build
#14 4.033 creating build/lib.linux-x86_64-3.8
#14 4.034 creating build/lib.linux-x86_64-3.8/lupa
#14 4.034 copying lupa/__init__.py -> build/lib.linux-x86_64-3.8/lupa
#14 4.036 copying lupa/version.py -> build/lib.linux-x86_64-3.8/lupa
#14 4.037 running build_ext
#14 4.049 building 'lupa._lupa' extension
#14 4.049 creating build/temp.linux-x86_64-3.8
#14 4.050 creating build/temp.linux-x86_64-3.8/lupa-master
#14 4.050 creating build/temp.linux-x86_64-3.8/lupa-master/lupa
#14 4.050 gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DCYTHON_CLINE_IN_TRACEBACK=0 -I../LuaJIT-2.1.0-beta3/src -I/usr/local/include/python3.8 -c /lupa-master/lupa/_lupa.c -o build/temp.linux-x86_64-3.8/lupa-master/lupa/_lupa.o
#14 13.09 gcc -pthread -shared -Wl,--strip-all build/temp.linux-x86_64-3.8/lupa-master/lupa/_lupa.o ../LuaJIT-2.1.0-beta3/src/libluajit.a -L/usr/local/lib -o build/lib.linux-x86_64-3.8/lupa/_lupa.cpython-38-x86_64-linux-gnu.so
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_err.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_obj.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_str.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_func.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_debug.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_state.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_dispatch.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_strfmt.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_strfmt_num.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_api.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_profile.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_lex.o)' is incompatible with i386:x86-64 output
#14 13.33 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_parse.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_load.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_trace.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ctype.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ccallback.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_alloc.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_aux.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_ffi.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_init.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vm.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_gc.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_char.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_bc.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_buf.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_tab.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_udata.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_meta.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmevent.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_strscan.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_bcread.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_bcwrite.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_opt_dce.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_opt_loop.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_opt_split.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_opt_sink.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_mcode.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_snap.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_record.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ffrecord.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_asm.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_cdata.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_cconv.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ccall.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_clib.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_cparse.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_lib.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_base.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_math.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_bit.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_string.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_table.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_io.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_os.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_package.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_debug.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_jit.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_opt_mem.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_opt_fold.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_opt_narrow.o)' is incompatible with i386:x86-64 output
#14 13.34 /usr/bin/ld: i386 architecture of input file `../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_crecord.o)' is incompatible with i386:x86-64 output
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_strfmt.o): in function `lj_strfmt_putfxint':
#14 13.35 lj_strfmt.c:(.text+0x116c): undefined reference to `__umoddi3'
#14 13.35 /usr/bin/ld: lj_strfmt.c:(.text+0x118f): undefined reference to `__udivdi3'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ctype.o): in function `lj_ctype_repr_int64':
#14 13.35 lj_ctype.c:(.text+0x16fc): undefined reference to `__umoddi3'
#14 13.35 /usr/bin/ld: lj_ctype.c:(.text+0x1723): undefined reference to `__udivdi3'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_log':
#14 13.35 lj_vmmath.c:(.text+0x1e): undefined reference to `log'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_log10':
#14 13.35 lj_vmmath.c:(.text+0x4e): undefined reference to `log10'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_exp':
#14 13.35 lj_vmmath.c:(.text+0x7e): undefined reference to `exp'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_sin':
#14 13.35 lj_vmmath.c:(.text+0xae): undefined reference to `sin'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_cos':
#14 13.35 lj_vmmath.c:(.text+0xde): undefined reference to `cos'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_tan':
#14 13.35 lj_vmmath.c:(.text+0x10e): undefined reference to `tan'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_asin':
#14 13.35 lj_vmmath.c:(.text+0x13e): undefined reference to `asin'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_acos':
#14 13.35 lj_vmmath.c:(.text+0x16e): undefined reference to `acos'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_atan':
#14 13.35 lj_vmmath.c:(.text+0x19e): undefined reference to `atan'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_sinh':
#14 13.35 lj_vmmath.c:(.text+0x1ce): undefined reference to `sinh'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_cosh':
#14 13.35 lj_vmmath.c:(.text+0x1fe): undefined reference to `cosh'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_tanh':
#14 13.35 lj_vmmath.c:(.text+0x22e): undefined reference to `tanh'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_atan2':
#14 13.35 lj_vmmath.c:(.text+0x26a): undefined reference to `atan2'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_pow':
#14 13.35 lj_vmmath.c:(.text+0x2aa): undefined reference to `pow'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `lj_wrap_fmod':
#14 13.35 lj_vmmath.c:(.text+0x2ea): undefined reference to `fmod'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L41':
#14 13.35 lj_vmmath.c:(.text+0x3bc): undefined reference to `pow'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L38':
#14 13.35 lj_vmmath.c:(.text+0x3fc): undefined reference to `atan2'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L68':
#14 13.35 lj_vmmath.c:(.text+0x4e6): undefined reference to `cos'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L66':
#14 13.35 lj_vmmath.c:(.text+0x4fe): undefined reference to `tan'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L74':
#14 13.35 lj_vmmath.c:(.text+0x576): undefined reference to `exp'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L73':
#14 13.35 lj_vmmath.c:(.text+0x586): undefined reference to `exp2'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L72':
#14 13.35 lj_vmmath.c:(.text+0x596): undefined reference to `log'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L71':
#14 13.35 lj_vmmath.c:(.text+0x5a6): undefined reference to `log2'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L70':
#14 13.35 lj_vmmath.c:(.text+0x5b6): undefined reference to `log10'
#14 13.35 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_vmmath.o): in function `.L69':
#14 13.35 lj_vmmath.c:(.text+0x5c6): undefined reference to `sin'
#14 13.35 /usr/bin/ld: lj_vmmath.c:(.text+0x5d5): undefined reference to `sqrt'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o): in function `lj_carith_divu64':
#14 13.36 lj_carith.c:(.text+0x372): undefined reference to `__udivdi3'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o): in function `lj_carith_divi64':
#14 13.36 lj_carith.c:(.text+0x41b): undefined reference to `__divdi3'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o): in function `lj_carith_modu64':
#14 13.36 lj_carith.c:(.text+0x492): undefined reference to `__umoddi3'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o): in function `lj_carith_modi64':
#14 13.36 lj_carith.c:(.text+0x566): undefined reference to `__moddi3'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o): in function `.L170':
#14 13.36 lj_carith.c:(.text+0x139b): undefined reference to `__umoddi3'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o): in function `.L171':
#14 13.36 lj_carith.c:(.text+0x13e9): undefined reference to `__udivdi3'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_carith.o): in function `.L173':
#14 13.36 lj_carith.c:(.text+0x15ea): undefined reference to `__moddi3'
#14 13.36 /usr/bin/ld: lj_carith.c:(.text+0x162a): undefined reference to `__divdi3'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_clib.o): in function `clib_error_':
#14 13.36 lj_clib.c:(.text+0x13): undefined reference to `dlerror'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_clib.o): in function `lj_clib_index':
#14 13.36 lj_clib.c:(.text+0x1cc): undefined reference to `dlsym'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_clib.o): in function `lj_clib_load':
#14 13.36 lj_clib.c:(.text+0x309): undefined reference to `dlopen'
#14 13.36 /usr/bin/ld: lj_clib.c:(.text+0x3d9): undefined reference to `dlerror'
#14 13.36 /usr/bin/ld: lj_clib.c:(.text+0x488): undefined reference to `dlopen'
#14 13.36 /usr/bin/ld: lj_clib.c:(.text+0x497): undefined reference to `dlerror'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_clib.o): in function `lj_clib_unload':
#14 13.36 lj_clib.c:(.text+0x594): undefined reference to `dlclose'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_math.o): in function `lj_ffh_math_log':
#14 13.36 lib_math.c:(.text+0x123): undefined reference to `log2'
#14 13.36 /usr/bin/ld: lib_math.c:(.text+0x137): undefined reference to `log2'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_package.o): in function `ll_loadfunc':
#14 13.36 lib_package.c:(.text+0x949): undefined reference to `dlsym'
#14 13.36 /usr/bin/ld: lib_package.c:(.text+0x991): undefined reference to `dlsym'
#14 13.36 /usr/bin/ld: lib_package.c:(.text+0x99a): undefined reference to `dlerror'
#14 13.36 /usr/bin/ld: lib_package.c:(.text+0x9c3): undefined reference to `dlsym'
#14 13.36 /usr/bin/ld: lib_package.c:(.text+0xaaf): undefined reference to `dlopen'
#14 13.36 /usr/bin/ld: lib_package.c:(.text+0xae1): undefined reference to `dlerror'
#14 13.36 /usr/bin/ld: lib_package.c:(.text+0xb02): undefined reference to `dlerror'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_package.o): in function `lj_cf_package_loader_preload':
#14 13.36 lib_package.c:(.text+0xd2b): undefined reference to `dlsym'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lib_package.o): in function `lj_cf_package_unloadlib':
#14 13.36 lib_package.c:(.text+0x1041): undefined reference to `dlclose'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x120): undefined reference to `sinh'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x128): undefined reference to `cosh'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x130): undefined reference to `tanh'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x170): undefined reference to `exp'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x178): undefined reference to `exp2'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x180): undefined reference to `log'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x188): undefined reference to `log2'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x190): undefined reference to `log10'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x198): undefined reference to `sin'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x1a0): undefined reference to `cos'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x1a8): undefined reference to `tan'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x1b8): undefined reference to `pow'
#14 13.36 /usr/bin/ld: ../LuaJIT-2.1.0-beta3/src/libluajit.a(lj_ir.o):(.data.rel.ro+0x1c0): undefined reference to `atan2'
#14 13.37 collect2: error: ld returned 1 exit status
#14 13.37 warning: lupa/_lupa.pyx:1719:5: Unraisable exception in function 'lupa._lupa.py_object_gc_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:1791:5: Unraisable exception in function 'lupa._lupa.py_call_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:1817:5: Unraisable exception in function 'lupa._lupa.py_str_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:1895:5: Unraisable exception in function 'lupa._lupa.py_object_getindex_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:1915:5: Unraisable exception in function 'lupa._lupa.py_object_setindex_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:1967:5: Unraisable exception in function 'lupa._lupa.py_wrap_object_protocol_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:2037:5: Unraisable exception in function 'lupa._lupa.py_enumerate_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:2047:5: Unraisable exception in function 'lupa._lupa.py_iter_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:2088:5: Unraisable exception in function 'lupa._lupa.py_iter_next_with_gil'.
#14 13.37 warning: lupa/_lupa.pyx:2125:5: Unraisable exception in function 'lupa._lupa.py_args_with_gil'.
#14 13.37 error: command 'gcc' failed with exit status 1
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code = Unknown desc = failed to build LLB: executor failed running [/bin/sh -c cd lupa-master     && python setup.py bdist_wheel --lua-includes ../LuaJIT-2.1.0-beta3/src --lua-lib ../LuaJIT-2.1.0-beta3/src/libluajit.a]: runc did not terminate sucessfully
```
</details>
</summary>

<summary>
And this is the Dockerfile I'm using.
<details>

```Dockerfile
# syntax = docker/dockerfile:1.2
FROM python:3.8-slim

# Setup cache mount
ENV PIP_CACHE_DIR=/var/cache/pip
RUN mkdir -p $PIP_CACHE_DIR && rm -f /etc/apt/apt.conf.d/docker-clean

# Install basic tools
RUN --mount=type=cache,target=/var/cache/apt \
    apt-get update \
    && apt-get install -y --no-install-recommends \
    unzip \
    gzip \
    curl \
    ca-certificates \
    && rm -rf /var/lib/apt/lists/*

# Get LuaJIT
RUN curl http://luajit.org/download/LuaJIT-2.1.0-beta3.tar.gz -sLo luajit.tar.gz \
    && tar -xvzf luajit.tar.gz \
    && rm luajit.tar.gz

# Get lupa
RUN curl https://github.com/scoder/lupa/archive/refs/heads/master.zip -sLo lupa.zip \
    && unzip lupa.zip \
    && rm lupa.zip

# Install tools required for build luajit
RUN --mount=type=cache,target=/var/cache/apt \
    apt-get update \
    && apt-get install -y --no-install-recommends \
    build-essential \
    gcc \
    gcc-multilib \
    make \
    && rm -rf /var/lib/apt/lists/*

# Install tools required for build lupa wheel
RUN pip install -U pip \
    && pip install Cython setuptools wheel

# Build luajit
RUN cd LuaJIT-2.1.0-beta3/src \
    && make CC="gcc -m32" CFLAGS="-fPIC"

# Build lupa
RUN cd lupa-master \
    && python setup.py bdist_wheel --lua-includes ../LuaJIT-2.1.0-beta3/src --lua-lib ../LuaJIT-2.1.0-beta3/src/libluajit.a
```

</details>
</summary>

[an answer in SO](https://stackoverflow.com/questions/19200333/architecture-of-i386-input-file-is-incompatible-with-i386x86-64) says it can be fixed by `ld -m elf_i386 -s -o file file.o`, but i can't find a proper way to add the args to ld.

Is there any way to fix this? or any place I can get a prebuilt lupa with 32bit luajit?

**Comment by scoder** [comment]
I don't think this can be done. The SO answer talks about building 32bit applications on a 64bit system. Not integrating a 32bit library into a 64bit application. You need a 32bit Python for this.

**Comment by AutumnSun1996** [comment]
It is possible.
I managed to build a wheel `lupa-1.10-cp38-cp38-linux_x86_64.whl` with 32bit luajit.
It's quite a long time ago, so I've forgot how I did it, but there exists such a wheel that I use to run 32bit luajit bytecode in lupa.

Btw, I also tried for Windows even earlier, and automated it in https://github.com/AzurLaneTools/LupaWinBuilds/

Anyway, the issue is already solved, so it's ok to keep it closed.
